src/energyplus/ReverseTranslator/ReverseTranslateSurfaceControlMovableInsulation.cpp

```diff
@@ -0,0 +1,110 @@
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "../ReverseTranslator.hpp"
+
+#include "../../model/SurfaceControlMovableInsulation.hpp"
+#include "../../model/SurfaceControlMovableInsulation_Impl.hpp"
+#include "../../model/Surface.hpp"
+#include "../../model/Surface_Impl.hpp"
+#include "../../model/Material.hpp"
+#include "../../model/Material_Impl.hpp"
+#include "../../model/Schedule.hpp"
+#include "../../model/Schedule_Impl.hpp"
+
+#include "../../utilities/idf/WorkspaceExtensibleGroup.hpp"
+
+#include <utilities/idd/SurfaceControl_MovableInsulation_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+
+#include "../../utilities/core/Assert.hpp"
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+OptionalModelObject ReverseTranslator::translateSurfaceControlMovableInsulation( const WorkspaceObject & workspaceObject )
+{
+  if( workspaceObject.iddObject().type() != IddObjectType::SurfaceControl_MovableInsulation ){
+    LOG(Error, "WorkspaceObject is not IddObjectType: SurfaceControlMovableInsulation");
+    return boost::none;
+  }
+
+  OptionalString insulationType = workspaceObject.getString(openstudio::SurfaceControl_MovableInsulationFields::InsulationType);
+  OptionalWorkspaceObject target;
+
+  boost::optional<Surface> surface;
+  target = workspaceObject.getTarget(openstudio::SurfaceControl_MovableInsulationFields::SurfaceName);
+  if (target){
+    OptionalModelObject modelObject = translateAndMapWorkspaceObject(*target);
+    if (modelObject){
+      if (modelObject->optionalCast<Surface>()){
+        surface = modelObject->cast<Surface>();
+      }
+    }
+  }
+
+  boost::optional<Material> material;
+  target = workspaceObject.getTarget(openstudio::SurfaceControl_MovableInsulationFields::MaterialName);
+  if (target){
+    OptionalModelObject modelObject = translateAndMapWorkspaceObject(*target);
+    if (modelObject){
+      if (modelObject->optionalCast<Material>()){
+        material = modelObject->cast<Material>();
```

Comment:
Either `   surfaceControlMovableInsulation.setMaterial(*material); ` gets set here.

But if not allowing a broken state, then you do want to both check if Surface exists and material too.

Then 

```c++
if (!surface.has_value() || !material.has_value()) {
 // LOG_AND_THROW or return
}

SurfaceControlMovableInsulation surfaceControlMovableInsulation(*surface, *material);
```